hydra-task/src/main/java/com/addthis/hydra/task/source/DataSourceQuery.java

```diff
@@ -0,0 +1,192 @@
+
+import java.io.File;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import java.nio.file.Paths;
+
+import com.addthis.basis.util.Files;
+
+import com.addthis.bundle.channel.DataChannelOutput;
+import com.addthis.bundle.core.Bundle;
+import com.addthis.bundle.core.BundleField;
+import com.addthis.bundle.core.BundleFormat;
+import com.addthis.bundle.core.list.ListBundleFormat;
+import com.addthis.bundle.util.AutoField;
+import com.addthis.hydra.data.query.Query;
+import com.addthis.hydra.data.query.QueryOpProcessor;
+import com.addthis.hydra.data.query.source.QueryEngineSource;
+import com.addthis.hydra.data.query.source.QueryEngineSourceSingle;
+import com.addthis.hydra.data.tree.ReadTree;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.FluentIterable;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DataSourceQuery extends AbstractStreamDataSource implements DataChannelOutput {
+
+    private static final Logger log = LoggerFactory.getLogger(DataSourceQuery.class);
+
+    private final BundleFormat formatIn = new ListBundleFormat();
+
+    private final BundleFormat formatOut = new ListBundleFormat();
+
+    /**
+     * The tree job that is to be queried. Specify no jobId
+     * for a job to query itself. This field is optional.
+     */
+    @JsonProperty private String jobId;
+
+    /**
+     * Optionally specify the directory
+     * of the tree for the given job. Default
+     * is "data".
+     */
+    @JsonProperty(required = true) private String treeDir;
+
+    /**
+     * Path for the query. This field is required.
+     */
+    @JsonProperty(required = true) private String path;
+
+    /**
+     * Operations for the query. This field is optional.
+     */
+    @JsonProperty private String ops;
+
+    /**
+     * Optionally specify the field names for the columns
+     * produced by the query. Otherwise the fields will
+     * be named "0", "1", "2", etc. Default is null.
+     */
+    @JsonProperty private AutoField[] fields;
+
+    private final AtomicReference<Throwable> firstError = new AtomicReference<>();
+
+    private QueryEngineSourceSingle queryEngine;
+
+    private QueryEngineSource.Handle queryHandle;
+
+    private File tempDir;
+
+    private boolean localInputSource;
+
+    @Override public void init() {
+        super.init();
+        localInputSource = (jobId == null);
+        if (!localInputSource) {
+            throw new UnsupportedOperationException("At this time only local queries are supported");
+        }
+        try {
+            File dir = Paths.get(treeDir).toFile();
+            tempDir = Files.createTempDir();
+            queryEngine = new QueryEngineSourceSingle(new ReadTree(dir));
+            Query query = new Query(System.getenv("HYDRA_JOBID"), new String[] {path}, null);
```

Comment:
then you can have another field or a TaskRunConfig object. Directly accessing environment variables something we would like to avoid. It is the equivalent of a static field